internal/pkg/agent/application/dispatcher/dispatcher_test.go

```diff
@@ -484,3 +489,85 @@ func Test_ActionDispatcher_scheduleRetry(t *testing.T) {
 		action.AssertExpectations(t)
 	})
 }
+
+func TestReportNextScheduledUpgrade(t *testing.T) {
+	now := time.Now().UTC()
+	later := now.Add(3 * time.Hour)
+	muchLater := later.Add(3 * time.Hour)
+
+	cases := map[string]struct {
+		actions               []fleetapi.Action
+		expectedDetails       *details.Details
+		expectedWarningLogMsg string
+	}{
+		"no_scheduled_upgrades": {
```

Comment:
add a testcase with upgrade start time being some unparsable time

Reply:
Added in f31763a063.